lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -339,12 +345,15 @@ def init_api_client
         key = Google::APIClient::PKCS12.load_key(@private_key_path,
                                                  @private_key_passphrase)
         jwt_asserter = Google::APIClient::JWTAsserter.new(
-          @private_key_email, 'https://www.googleapis.com/auth/logging.write',
-          key)
+          @private_key_email, LOGGING_SCOPE, key)
         @client.authorization = jwt_asserter.to_authorization
         @client.authorization.expiry = 3600  # 3600s is the max allowed value
       else
-        @client.authorization = Google::APIClient::ComputeServiceAccount.new
+        if !@application_credentials_path.nil?
+          ENV['GOOGLE_APPLICATION_CREDENTIALS'] = @application_credentials_path
+        end
+        @client.authorization = Google::Auth.get_application_default(
+            LOGGING_SCOPE)
```

Comment:
Correct - see https://developers.google.com/identity/protocols/application-default-credentials
Unfortunately this is the only mechanism the googleauth API allows for specifying a different file to use.

FWIW, I'm a bit on the fence about whether we should have this configuration option at all.  The user could also edit /etc/init.d/google-fluentd and add 'export GOOGLE_APPLICATION_CREDENTIALS=<xxx>' instead.  That's really no more or less difficult to do (or document) than putting it in the config file, and since it's the "canonical" way to do it perhaps we should do that rather than giving the user two ways to do the same thing?